### DISCUSSION TITLE
Property 'getInitialProps' does not exist on type 'FC<ItemProps>'.ts(2339)

### DISCUSSION THREAD

**Discussion by TheoMer** [vercel/next.js#17330] [Help] [answered]
react: "^16.12.0",   
react-dom: "^16.12.0",   
@types/react: "^16.9.49",   
typescript: "^4.0.3",
"next": "^9.4.4",

How doI resolve the above mentioned issue? My code is as follows:

```
import React, {FC} from "react";
import SingleItem from '../components/SingleItem';
import { useRouter } from 'next/router';

interface ItemProps{
  query:{
      id:string;
  };
  userIP:string;
  userAgent:string;
  urlReferer:string;
}

const Item:FC<ItemProps> = (props) => {
  const router = useRouter();
  return(
    <div>
      <style dangerouslySetInnerHTML={{ __html: SingleItemStyles }} />
      <SingleItem id={props.query.id} user_ip={props.userIP} user_Agent={props.userAgent} url={router.asPath} urlReferer={props.urlReferer} />
    </div>
  );
}

// Obtain User IP on initial site load
Item.getInitialProps = async ({ req }) => {
  const userIP = req ? (req.headers['x-real-ip'] || req.connection.remoteAddress) : false;
  const userAgent = req ? req.headers['user-agent'] : navigator.userAgent;
  const urlReferer = req ? req.headers['referer'] : document.referrer;
  
  return { userIP, userAgent, urlReferer }
}

export default Item;
```

**Comment by ljosberinn** [comment, marked as answer]
[The docs mention this](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps#typescript), which should always be your first stop.

**Reply by TheoMer** [reply]
Hey @ljosberinn many thanks for the docs link.

**Reply by Damdias** [reply]
easy solution declare type
declare global {
  namespace React {
    interface FunctionComponent<P = {}> {
      getInitialProps(): void;
    }
  }
}

**Comment by anjan-kp** [comment]
@ljosberinn I am using latest nextJS and facing problem if I add getInitialProps in my `_app.tsx` file.
It's giving the same error. As per your recommended DOC I extended MyApp to NextPage.
But I got error on `Component and pageProps`. My code looks like : `const MyApp: FC<AppProps> = ({ Component, pageProps }) => {`
Thanks in advance!
